src/commands/buildJs.ts

```diff
@@ -142,25 +144,41 @@ async function waitAllAndThrowIfAnyCompilationsFailed(
         };
       }
       // has some errors
-      const { message: stderr } = result.reason as CompilerError;
-      const [command, , ...messages] = stderr.split("\n");
+      const reason = result.reason as CompilerError;
       try {
-        const items: CompileErrorItem[] = JSON.parse(messages.join("\n"));
+        const { command, items } = parseErrorReason(reason, config);
         return {
           entryConfigPath: result.entryConfigPath,
           command,
           items
         };
       } catch {
         // for invalid compiler options errors
-        throw new Error(`Unexpected non-JSON error: ${stderr}`);
+        throw new Error(`Unexpected non-JSON error: ${reason.message}`);
       }
     })
     .filter(result => result.items.length > 0);
   if (results.filter(result => result.isRejected).length > 0) {
     throw new BuildJsCompilationError(reasons, results.length);
   }
   return reasons;
+
+  function parseErrorReason(
+    reason: CompilerError & { info?: Record<string, any> },
+    config: DuckConfig
+  ): { command: string; items: CompileErrorItem[] } {
+    const { message: stderr, info } = reason;
+    const message = isBatchMode(config.batch) && info ? info.message : stderr;
```

Comment:
@shisama 
According to https://github.com/cybozu/duck/pull/378#discussion_r378350303, this issue would occur when `duck` use faastjs to build JS.

duck decides whether to use faastjs or not by the following condition.
https://github.com/cybozu/duck/blob/b8e1583d81489ebaa3f0c51a13550f9f143ac4ab/src/commands/buildJs.ts#L37
So I guess that you can use the same condition for this.

Reply:
@koba04 
Thank you. I fixed it. If you think that we don't need to check if 'info' exists, please tell me again.